src/RustyObjectStore.jl

```diff
@@ -1755,6 +1903,54 @@ function finish!(stream::ListStream)
     return true
 end
 
+mutable struct StageInfoResponseFFI
+    result::Cint
+    stage_info::Ptr{Cchar}
+    error_message::Ptr{Cchar}
+    context::Ptr{Cvoid}
+
+    StageInfoResponseFFI() = new(-1, C_NULL, C_NULL, C_NULL)
+end
+
+function current_stage_info(conf::AbstractConfig)
+    response = StageInfoResponseFFI()
+    ct = current_task()
+    event = Base.Event()
+    handle = pointer_from_objref(event)
+    config = into_config(conf)
+    while true
+        preserve_task(ct)
+        result = GC.@preserve config response event try
+            result = @ccall rust_lib.current_stage_info(
+                config::Ref{Config},
+                response::Ref{StageInfoResponseFFI},
+                handle::Ptr{Cvoid}
+            )::Cint
+
+            wait_or_cancel(event, response)
+
+            result
+        finally
+            unpreserve_task(ct)
+        end
+
+        if result == 2
+            # backoff
+            sleep(0.01)
+            continue
+        end
+
+        # No need to destroy_write_stream in case of errors here
+        @throw_on_error(response, "current_stage_info", GetException)
```

Comment:
I thought I had forgotten to free it here, but in reality it is as you say, if the error is set it is guaranteed that there was no allocation. The problem is really the comment above, that says `destroy_write_stream` instead of `destroy_cstring(response.stage_info)`

Reply:
Done